### PR TITLE
bazel: build/test almost everything

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: Miscellaneous cleanups and bug fixes. The main motivating idea here was to make `bazel build //...` and `bazel test //...` mostly work. (There's a few reasons these still don't work, but we're a lot closer.)

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/assign @BenTheElder @mikedanese @spxtr

### CODE REVIEW

**Comment by ixdy** [comment]
The automanaged/licenses commits are further cleanup from #50407, as some of them crept back in from PRs that were in flight.

**Comment by ixdy** [comment]
/assign @thockin 
(for top-level approval of BUILD file changes)

As noted in https://github.com/kubernetes/kubernetes/pull/52873#issuecomment-331305328, most of this is cleanup from PRs-in-flight from a previous full-repo cleanup.

File: build/root/BUILD.root [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 package(default_visibility = ["//visibility:public"])
 
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_prefix")
+load("@io_bazel_rules_go//go:def.bzl", "go_prefix")
```
Comment (dhilipkumars) [inline build/root/BUILD.root]: Hi @ixdy, 

Does this PR automatically fix `./hack/update-bazel.sh`, this seem to include `go_library` back.

Comment (ixdy) [reply build/root/BUILD.root]: it shouldn't be doing that. Do you have a `generated.pb.go` and/or `types.go` file in the root of your kubernetes tree? I wonder if one of the code generation steps is interacting poorly.

**Pull request merged**